litmus-portal/backend/auth/controller/controller.go

```diff
@@ -14,7 +14,8 @@ type UserController interface {
 	Update(c *gin.Context)
 }
 
-// UpdateController will do update operation
-type UpdateController interface {
-	Update(c *gin.Context)
+// GithubController will do login operation
+type GithubController interface {
```

Comment:
can there be an interface for OAuth and anyone can plug, github, google, shibboleth, etc.